src/conda_package_handling/archive_utils_c.c

```diff
@@ -0,0 +1,186 @@
+// Utilities for creating and extracting files from archives using libarchive
+// for use in conda-packaging-handling
+
+#include <fcntl.h>
+#include <archive.h>
+#include <archive_entry.h>
+
+struct archive * prepare_gnutar_archive(
+    const char *outname, const char *filtername, const char *opts, const char **err_str)
+{
+	struct archive *a;
```

Comment:
Same here tab vs 4 spaces ...

Reply:
Also you should verify early that err_str is not NULL: Something like:
   if (!err_str)
     return NULL;